sklearn/metrics/ranking.py

```diff
@@ -255,9 +270,33 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
                                         sample_weight=sample_weight)
         return auc(fpr, tpr, reorder=True)
 
-    return _average_binary_score(
-        _binary_roc_auc_score, y_true, y_score, average,
-        sample_weight=sample_weight)
+    if type_of_target(y_true) != "multiclass":
+        return _average_binary_score(
+            _binary_roc_auc_score, y_true, y_score, average,
+            sample_weight=sample_weight)
+    else:
+        # validation for multiclass parameter specifications
+        average_options = ("macro", "weighted")
+        if average not in average_options:
+            raise ValueError("average has to be one of {0}"
+                             "".format(average_options))
+        multiclass_options = ("ovo", "ovr")
+        if multiclass not in multiclass_options:
+            raise ValueError("'{0}' is not supported for multiclass ROC AUC"
+                             "".format(multiclass))
+
+        check_consistent_length(y_true, y_score)
+        check_array(y_true, ensure_2d=False)
+        check_array(y_score)
+
+        if multiclass == "ovo":
```

Comment:
The approaches here do not support `sample_weight` currently. If you maintain that, you should raise an error if `sample_weight is not None`.